src/GraphQL/Validation/BasicVisitor.cs

```diff
@@ -38,7 +38,8 @@ public void Visit(INode node)
 
             for (int i = 0; i < _visitors.Count; i++)
             {
-                _visitors[i].Enter(node);
+                if (_visitors[i].ShouldRunOn(context))
```

Comment:
Within the document validator, you're filtering by ShouldRunOn, so it's not needed here, correct?

Reply:
I know. But theoretically this class can be used on its own.